### PR TITLE
[BABEL-949] Support NEXT VALUE FOR

### PR DESCRIPTION
**Pull request by timchang514**
### Description

Adding initial support for NEXT VALUE FOR, typically used to get the next value of a sequence. We don't support NEXT VALUE FOR in control-of-flow statements or the OVER clause yet in this change. 

This change was added because it is a feature that customers use (for example, in table declarations as a default value).

### Issues Resolved

BABEL-949

### Test Scenarios Covered ###
* **Use case based -**
Unit tests added. 

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 1]
```diff
@@ -1857,6 +1857,24 @@ class tsqlBuilder : public tsqlCommonMutator
 	void exitFunction_call(TSqlParser::Function_callContext *ctx) override
 	{
 		is_function = true;
+		if (ctx->NEXT())
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp]: `exitFunction_call` is called not only for CREATE TABLE but also in other cases. Let's add more test cases other than CREATE TABLE case.

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 2]
```diff
@@ -1857,6 +1857,24 @@ class tsqlBuilder : public tsqlCommonMutator
...
+		{
+			std::string seq_name;
+			if (ctx->full_object_name())
+			{
+				seq_name = ::getFullText(ctx->full_object_name());
+			}
+			else
+			{
+				elog(WARNING, "placeholder");
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp]: What is this for and what happens if `ctx->full_object_name()` is null? Is it safe to proceed?

File: test/JDBC/expected/BABEL-949.out [hunk 3]
```diff
@@ -0,0 +1,36 @@
+CREATE SEQUENCE seq_949 AS INT START WITH 1 INCREMENT BY 1;
```
Comment (2jungkook) [inline test/JDBC/expected/BABEL-949.out]: 1. Let's try to use more descriptive test file name, e.g., `BABEL-NEXT-VALUE-FOR`.
2. Please make this as an upgrade test and put it in the latest and 14_7 schedule files.
3. The prepare script would be exactly same as this file and in the verification script, you need to further check INSERT and SELECT NEXT VALUE.

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 4]
```diff
@@ -1857,6 +1857,16 @@ class tsqlBuilder : public tsqlCommonMutator
 	void exitFunction_call(TSqlParser::Function_callContext *ctx) override
 	{
 		is_function = true;
+		if (ctx->NEXT() && ctx->full_object_name())
+		{
+			std::string seq_name = ::getFullText(ctx->full_object_name());
+			std::string nextval_string = "nextval('" + seq_name + "')";
+
+			rewritten_query_fragment.emplace(std::make_pair(ctx->NEXT()->getSymbol()->getStartIndex(), std::make_pair(::getFullText(ctx->NEXT()), "")));
```
Comment (macwoody) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp:1873]: could you please add a test for schema.seq case?

**Review by macwoody — APPROVED** [review]

**Pull request merged**